### PR TITLE
update TS interface

### LINKED ISSUE #2136: TS interface AXAButtonProps lacks event argument for onClick function
## Expected Behavior
When using the AXAButton Component  it should be possible to pass an event as argument to the onClick function.

## Current Behavior
When using the AXAButton Component with TypeScript, it is not possible to access the event in the provided onClick function, 
without making use of @ts-ignore because the onClick function attribute in the interface AXAButtonProps does not allow any arguments (such as an event).

## Steps to Reproduce
```
<AXAButton
  type='submit'
  variant='red'
  // @ts-ignore --> HERE WE NEED TO USE @ts-ignore because the onClick function does not allow arguments.
  onClick={e => {
    e.preventDefault();
    fooBar();
  }}
>
  suchen
</AXAButton>
```

## Context (Environment)
We want to prevent the defualt event of an submit button.

## Possible Solution
Add an event argument to the onClick function on interface AXAButtonProps.

### PR DESCRIPTION
**Pull request by danyball**
Fixes #2136

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [x] Tests are sufficient.
- [x] Modifications still work in IE.
- [x] Documentation is up to date.
- [x] Changelog is up to date.
- [x] Typescript typings for properties are up to date.
- [x] Designers approved changes or no approval needed.
- [x] Dependencies to other components still work.

### CODE REVIEW

File: src/components/10-atoms/button/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 6.0.16
+
+- Add the event to the typescript interface. #2136
```
Comment (markus-walther) [inline src/components/10-atoms/button/CHANGELOG.md]: Wording improvement: Add event to onClick type description for TypeScript users

File: src/components/10-atoms/button/index.react.d.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export interface AXAButtonProps {
   size?: Size;
   motionOff?: boolean;
   disabled?: boolean;
-  onClick?: () => void;
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
```
Comment (markus-walther) [inline src/components/10-atoms/button/index.react.d.ts:23]: I trust you on this being correct ;-)

**Review by markus-walther — CHANGES_REQUESTED** [review]
Tiny wording improvement, looks good otherwise!

File: src/components/10-atoms/button/index.react.d.ts (hunk 2 above)
Comment (danyball) [reply src/components/10-atoms/button/index.react.d.ts:23]: We have exact this syntax at other components.

**Pull request merged**